File: common/library-sdk/src/main/java/com/dn/sdk/count/ITrack.java
package com.dn.sdk.count;

/**
 * @author by SnowDragon
 * Date on 2020/11/20
 * Description:
 */
public interface ITrack {

    /**
     * 统计广告点击
     */
    void onClick();

    /**
     * 统计广告展示
     */
    void onShow();

    /**
     * 统计广告关闭
     */
    void onAdClose();

    /**
     * 统计广告加载失败
     */
    void onLoadError();

    /**
     * 统计广告曝光成功
     */
    void onADExposure();

    /**
     * @param 是否给予奖励
     */
    void onRewardVerify(boolean b);


    /**
     * 视频播放完成
     */
    void onVideoComplete();


    /**
     * app下载完成
     */
    void downloadFinished();

}

File: common/library-sdk/src/main/java/com/dn/sdk/count/ITrack.kt
package com.dn.sdk.count

/**
 *
 * 广告事件追踪埋点事件
 *
 * @author XuShuai
 * @version v1.0
 * @date 2021/10/25 15:22
 */
interface ITrack {

    /**
     * 广告开始加载
     */
    fun onAdStartLoad()

    /**
     * 统计广告点击
     */
    fun onAdClick()

    /**
     * 统计广告展示
     */
    fun onAdShow()

    /**
     * 统计广告关闭
     */
    fun onAdClose()


    /**
     * @param verify 是否给予奖励
     */
    fun onRewardVerify(verify: Boolean)


    /**
     * 视频播放完成
     */
    fun onVideoComplete()

}